crates/bevy_ecs/src/query/filter.rs

```diff
@@ -93,6 +93,11 @@ pub unsafe trait QueryFilter: WorldQuery {
     ///
     /// Note that this is called after already restricting the matched [`Table`]s and [`Archetype`]s to the
     /// ones that are compatible with the Filter's access.
+    /// If the implementation of [`WorldQuery::matches_component_set`] is incorrect,
```

Comment:
Those 2 lines are a bit confusing to me; you mean that your `matches_component_set` was overly restrictive and was causing some tables/archetypes to be missed completely, which is why this function was not called?

Reply:
I think that this should be removed.